### PR TITLE
T2003

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f46af39d8520245/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f46af39d8520245/output.txt

Total script time: 0.58 mins

Server is running on http://pr796.butyr.org/templates/basic/
Run the Test Suite: http://pr796.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/485e7c5a058158e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/485e7c5a058158e/output.txt

Total script time: 0.61 mins

Server is running on http://pr796.butyr.org/templates/basic/
Run the Test Suite: http://pr796.butyr.org/test/

File: templates/saved-data.json [hunk 1]
```diff
@@ -41,25 +40,7 @@
...
-                "text": "lolz"
-              },
-              "track": "Layer1",
-              "name": "TrackEvent3"
-            }
-          ]        },
-        {
-          "name": "Layer2",
-          "id": "2",
+          "id": "Track2",
```
Comment (ScottDowne) [inline templates/saved-data.json:43]: Should be Layer2, not Track2, no?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/235773a5226816c/output.txt

File: src/core/track.js [hunk 2]
```diff
@@ -181,7 +182,13 @@ define( [
...
     }; //addTrackEvent
 
-    this.removeTrackEvent = function( trackEvent ){
+    /*
+     * Method removeTrackEvent
+     *
+     * @param {Object} trackEvent: The trackEvent to be removed from this track
+     * @param {Boolean} expectingTrackEvent: if true means we should not remove this track if it is empty as we are expecting a new trackEvent soon. This mostly comes in to play when adding/removing ghost trackEvents.
+     */
+    this.removeTrackEvent = function( trackEvent, expectingTrackEvent ){
```
Comment (jbuck) [inline src/core/track.js:191]: I :heart: you for doing this

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/235773a5226816c/output.txt

Total script time: 0.96 mins

File: templates/saved-data.json (hunk 1 above)
Comment (jbuck) [inline templates/saved-data.json:43]: I'd just revert all the changes made to this file; it's only used by the test template, which we don't care about anymore.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d06ed8bb16b7eb3/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d06ed8bb16b7eb3/output.txt

Total script time: 0.63 mins

Server is running on http://pr796.butyr.org/templates/basic/
Run the Test Suite: http://pr796.butyr.org/test/

**Comment by jbuck** [comment]
Staged on master: https://github.com/mozilla/butter/commit/bbc55eb5fff99fd13ec73560061bfd128549fbad

**Pull request closed without merging**